src/drivers/cms-client.ts

```diff
@@ -467,43 +467,63 @@ export class CMakeServerClient {
     log.warning(localize('cant.yet.handle.message', 'Can\'t yet handle the {0} messages', some.type));
   }
 
-  sendRequest(t: 'handshake', p: HandshakeParams): Promise<HandshakeContent>;
-  sendRequest(t: 'globalSettings', p?: GlobalSettingsParams): Promise<GlobalSettingsContent>;
-  sendRequest(t: 'setGlobalSettings', p: SetGlobalSettingsParams): Promise<SetGlobalSettingsContent>;
-  sendRequest(t: 'configure', p: ConfigureParams): Promise<ConfigureContent>;
-  sendRequest(t: 'compute', p?: ComputeParams): Promise<ComputeContent>;
-  sendRequest(t: 'codemodel', p?: CodeModelParams): Promise<CodeModelContent>;
-  sendRequest(t: 'cmakeInputs', p?: CMakeInputsParams): Promise<CMakeInputsContent>;
-  sendRequest(T: 'cache', p?: CacheParams): Promise<CacheContent>;
-  sendRequest(type: string, params: any = {}): Promise<any> {
-    const cp = {type, ...params};
-    const cookie = cp.cookie = Math.random().toString();
-    const pr = new Promise((resolve, reject) => { this._promisesResolvers.set(cookie, {resolve, reject}); });
-    const msg = JSON.stringify(cp);
+  private _sendRequest(type: 'handshake', params: HandshakeParams): Promise<HandshakeContent>;
+  private _sendRequest(type: 'globalSettings', params?: GlobalSettingsParams): Promise<GlobalSettingsContent>;
+  private _sendRequest(type: 'setGlobalSettings', params: SetGlobalSettingsParams): Promise<SetGlobalSettingsContent>;
+  private _sendRequest(type: 'configure', params: ConfigureParams): Promise<ConfigureContent>;
+  private _sendRequest(type: 'compute', params?: ComputeParams): Promise<ComputeContent>;
+  private _sendRequest(type: 'codemodel', params?: CodeModelParams): Promise<CodeModelContent>;
+  private _sendRequest(type: 'cmakeInputs', params?: CMakeInputsParams): Promise<CMakeInputsContent>;
+  private _sendRequest(type: 'cache', params?: CacheParams): Promise<CacheContent>;
+  private _sendRequest(type: string, params: any = {}): Promise<any> {
+    const cookiedMessage = {type, ...params};
+    const cookie = cookiedMessage.cookie = Math.random().toString();
+    const promise = new Promise((resolve, reject) => { this._promisesResolvers.set(cookie, {resolve, reject}); });
+    const jsonMessage = JSON.stringify(cookiedMessage);
     if (ENABLE_CMSERVER_PROTO_DEBUG) {
-      log.debug(localize('sending.message.to.cmake-server', 'Sending message to cmake-server: {0}', msg));
+      log.debug(localize('sending.message.to.cmake-server', 'Sending message to cmake-server: {0}', jsonMessage));
     }
     this._pipe.write('\n[== "CMake Server" ==[\n');
-    this._pipe.write(msg);
+    this._pipe.write(jsonMessage);
     this._pipe.write('\n]== "CMake Server" ==]\n');
-    return pr;
+    return promise;
   }
 
+  /**
```

Comment:
I propose in a separate commit to prepend a `request` or `do` (e.g. `requestConfigure`) to these methods to better express their purpose.